iina/MainWindowController.swift

```diff
@@ -1114,21 +1114,33 @@ class MainWindowController: PlayerWindowController {
         }
       }
     case .windowSize:
-      if fsState.isFullscreen { return }
+      if fsState.isFullscreen {
+        if recognizer.magnification < 0 {
+          self.toggleWindowFullScreen()
+          return
+        }
+      }
 
       // adjust window size
       if recognizer.state == .began {
         // began
         lastMagnification = recognizer.magnification
+        if window.frame.height >= screenFrame.height && lastMagnification > 0 {
```

Comment:
Looks like it works well if the video's aspect ratio is tall enough to reach the top of the screen first. But I think you should also be checking whether the width reaches the sides of the screen. In other words, try:
`if (window.frame.height >= screenFrame.height || window.frame.width >= screenFrame.width) && lastMagnification > 0 {`
Otherwise it won't work for very wide videos which may run out of screen width before running out of screen height.